### PR TITLE
[Improve] changed CollapsibleCard to functional component

### PR DESCRIPTION
**Pull request by aliseyalvi**
### What does this PR?
Changed CollapsibleCard component to functional component 
Used animation from react native reanimated Hooks
above improvement maybe fix the crashing issue on community screen

### Issue number
https://discord.com/channels/@me/920267778190086205/1095024906699292742
https://app.bugsnag.com/ecency/ecency-mobile/errors/640edfd3fb34b30008663e16?filters[event.unhandled]=true&filters[event.since]=30d&filters[app.release_stage]=production&filters[release.seen_in]=3.0.40%20(1679916584)&event_id=6433550f00bae5411cc70000

### Screenshots/Video

https://user-images.githubusercontent.com/48380998/231526045-dc2c884a-a254-4f4c-9387-a0f6e21e4de0.mp4

### CODE REVIEW

**Comment by noumantahir** [comment]
@aliseyalvi few things I noticed,

- when you accessed profile there was a slight open/close glitch, can make it smooth
- perhaps we should set community header to be expanded by default as well
- how would the header respond on content scroll.

**Comment by aliseyalvi** [comment]
@noumantahir 
 added autohide on scroll in profile screen and community screen

https://user-images.githubusercontent.com/48380998/232297476-a7fea82e-1f67-4bb4-a4df-afab42814cf0.mp4

File: src/screens/community/screen/communityScreen.js [hunk 1]
```diff
@@ -119,6 +132,7 @@ const CommunityScreen = ({ route }) => {
               selectedOptionIndex={_getSelectedIndex()}
               tag={tag}
               pageType="community"
+              handleOnScroll={isExpanded ? _handleOnScroll : null}
```
Comment (noumantahir) [inline src/screens/community/screen/communityScreen.js:135]: if simple enough, can we use onScrollBeingDrag to hide header as soon scroll begin instead of hiding at end of scroll, not sure why but handleOnScroll hiding header after I raise my finger and stop scrolling. will be great to have reveal animation on scroll back to top, but can be a separate PR etc.

**Review by noumantahir — CHANGES_REQUESTED** [review]
Apparently the glitch I was talking about is still there and is kind of annoying... check the video attached..

Another change regarding scroll to hide behavior, but that perhaps can be covered in a separate if need be, current behavior is also acceptable


https://user-images.githubusercontent.com/6298342/233486498-33e9f8ca-d8d4-4a86-be21-95d77e28a83d.mp4

**Comment by aliseyalvi** [comment]
@noumantahir PR is updated.

- collapse header when scroll begins drag in community screen
- collapse header when scroll begins drag in profile screen

Uploading Monosnap screencast 2023-05-06 21-11-31.mp4…

**Review by feruzm — APPROVED** [review]
